R/sparrow_integration.R

```diff
@@ -0,0 +1,262 @@
+##' pathways not targeted by a library will obviously never be enriched a positive target set regardless of
+##' their biological relevance, and pathways enriched within a focused library screen are similarly expected to partially
+##' reflect the composition of the library and other confounding issues (e.g., number of targets within a pathway).
+##' Analysts should therefore use this function with care. For example, it might be unsurprising to detect pathways related
+##' to histone modification within a screen employing a crispr library primarily targeting epigenetic regulators.
+##'  
+##' @param dflist A result object created by `ct.generateResults()`, or a named list containing many of them; will be passed as a list to 
+##' `ct.seasPrep()` with the associated `...` arguments.
+##' @param gdb A `GenseSetDb` object containing annotations for the targets specified in `result`.
+##' @param as.dfs Logical indicating whether to return the various contrast statistics as in-register lists of data.frames to facilitate comparisons.
+##' @param ... Additional arguments to pass to `ct.seasPrep()` or `sparrow::seas()`. 
+##' @return A named list of `SparrowResults` objects.
+##' @examples 
+##' data('resultsDF')
+##' ct.seas(list('longer' = resultsDF, 'shorter' = resultsDF[1:10000,]), gdb = getMSigGeneSetDb(collection = 'h', species = 'human', id.type = 'entrez'))
+##' @author Steve Lianoglou for seas; Russell Bainer for GeneSetDb processing and wrapping functions.
+##' @export
+ct.seas <- function(dflist,
+                    gdb, 
+                    as.dfs = FALSE,
+                    ...){
+  library('sparrow', quietly = TRUE)
+  #Check GSDB and determine feature set
+  stopifnot(is(gdb, 'GeneSetDb'), is(as.dfs, 'logical'))
+  
+  #listify results as needed
+  if(!is(dflist, 'list')){
+    if(ct.resultCheck(dflist)){
+      dflist <- list('result' = dflist)
+    }
+  }
+
+  
+  #Infer whether Gsdb is ID or feature centric
+  gids <- sum(gdb@featureIdMap$feature_id %in% dflist[[1]]$geneID)
+  gsids <- sum(gdb@featureIdMap$feature_id %in% dflist[[1]]$geneSymbol)
+  
+  if(all(c(gsids, gids) == 0)){
+    stop('None of the features in the GeneSetDb are present in either the geneID or geneSymbol slots of the first provided result.')
+  }
+  
+  identifier <- ifelse(gids > gsids, 'geneID', 'geneSymbol')
+  message(paste0('GeneSetDb feature_ids coded as ', identifier, 's.'))
+  if(identifier %in% 'geneID'){
+    message('Depending on the composition of your library, you might consider switching to a target-level analysis; see ?ct.GREATdb() for details.')
+    }
+  
+  #Check that all provided objects are keyed to the proper values
+  ipts <- ct.seasPrep(dflist, 
+                      collapse.on = identifier, 
+                      gdb = gdb)
+  
+  outs <- sapply(ipts, 
+                function(x){
+                  seas(gdb, x, methods = c('ora', 'fgsea'), rank_by = 'rank_by', selected = 'selected', groups = 'direction', ...)
```

Comment:
Heads up: I reordered the parameters in the `seas()` function to [put the "features" first](https://github.com/lianos/sparrow/commit/6744a683f08d73731a2212d30e38427f757bc309).

The only thing you need to do (after you update to sparrow 0.98.5) is to swap the order of `x` and `gdb` here, ie:

```r
seas(x, gdb, methods = c('ora', 'fgsea'), rank_by = 'rank_by', selected = 'selected', groups = 'direction', ...)
```

Sorry for the trouble, amigo.